src/main/java/ui/components/pickers/AssigneePickerDialog.java

```diff
@@ -0,0 +1,251 @@
+
+    private final List<PickerAssignee> originalAssignees = new ArrayList<>();
+    FlowPane assignedAssigneePane;
+    ScrollPane matchingAssigneePane;
+    VBox matchingAssigneesBox;
+    private TextField textField;
+    private AssigneePickerState state;
+
+    public AssigneePickerDialog(Stage stage, TurboIssue issue, List<TurboUser> assignees) {
+        initOwner(stage);
+        setTitle(DIALOG_TITLE);
+        setupButtons(getDialogPane());
+        setConfirmResultConverter();
+        originalAssignees.addAll(convertToPickerAssignees(issue, assignees));
+        state = new AssigneePickerState(originalAssignees);
+        initUI();
+        setupKeyEvents();
+
+        getExistingAssignee(originalAssignees)
+                .map(PickerAssignee::getLoginName)
+                .ifPresent(this::fillTextFieldWithAssigneeLoginName);
+    }
+
+    private void fillTextFieldWithAssigneeLoginName(String assigneeLoginName) {
+        textField.setText(assigneeLoginName);
+    }
+
+    private void setupKeyEvents() {
+        textField.textProperty().addListener((observable, oldValue, newValue) -> {
+            handleUpdatedInput(newValue);
+        });
+    }
+
+    private void handleUpdatedInput(String userInput) {
+        state = new AssigneePickerState(originalAssignees, userInput);
+        refreshUI(state);
+    }
+
+    private void handleMouseClick(String assigneeName) {
+        textField.setText(assigneeName);
+        refreshUI(state);
+    }
+
+    private List<PickerAssignee> convertToPickerAssignees(TurboIssue issue, List<TurboUser> assignees) {
+        List<PickerAssignee> originalAssignees = new ArrayList<>();
+        assignees.stream()
+        .forEach(assignee -> {
+            PickerAssignee convertedAssignee = new PickerAssignee(assignee);
+            if (isExistingAssignee(issue, convertedAssignee)) {
+                convertedAssignee.setExisting(true);
+            }
+            originalAssignees.add(convertedAssignee);
+        });
+
+        Collections.sort(originalAssignees);
+
+        return originalAssignees;
+    }
+
+    private boolean isExistingAssignee(TurboIssue issue, PickerAssignee assignee) {
+        if (!issue.getAssignee().isPresent()) return false;
+        return issue.getAssignee().get().equals(assignee.getLoginName());
+    }
+
+    private void setupButtons(DialogPane assigneePickerDialogPane) {
+        ButtonType confirmButtonType = new ButtonType("Confirm", ButtonBar.ButtonData.OK_DONE);
+        assigneePickerDialogPane.getButtonTypes().addAll(confirmButtonType, ButtonType.CANCEL);
+    }
+
+    private void setConfirmResultConverter() {
+        setResultConverter((dialogButton) -> {
+            List<PickerAssignee> finalList = state.getCurrentAssigneesList();
+            if (hasSelectedAssignee(finalList)) {
+                return new AssigneePickerDialogResponse(dialogButton,
+                        Optional.of(getSelectedAssignee(finalList).get().getLoginName()));
```

Comment:
This part can be `getSelectedAssignee(finalList).map(PickerAssignee::getLoginName)`.

Also, `finalList` can be `finalAssignees` or `suggestedAssignees` or something. Name can be more specific.